### PR TITLE
Vendor updated distribution for resumable downloads

### PR DESCRIPTION
**Pull request by aaronlehmann**
This vendors an updated version of `docker/distribution` that provides a HTTP reader which properly supports seeking via `Range` and `Content-Range` headers. It also updates local code for a few recent interface changes. Split out from #19840.

### CODE REVIEW

**Comment by calavera** [comment]
LGTM

File: distribution/pull_v2.go [hunk 1]
```diff
@@ -250,7 +250,7 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 	p.confirmedV2 = true
 
 	logrus.Debugf("Pulling ref from V2 registry: %s", ref.String())
-	progress.Message(p.config.ProgressOutput, tagOrDigest, "Pulling from "+p.repo.Name())
+	progress.Message(p.config.ProgressOutput, tagOrDigest, "Pulling from "+p.repo.Name().Name())
```
Comment (LK4D4) [inline distribution/pull_v2.go]: heh, `Name().Name()` looks weird. Maybe we should rename first `Name` method to something like `NamedRef`?

**Comment by dmcgowan** [comment]
LGTM

I am also :+1: for a follow up PR to rename the `Name()` in repository to something else. This would do away with some confusion with having `Name()` be a reference package defined function and how it is used in repository.

**Comment by BrianBland** [comment]
LGTM

**Comment by LK4D4** [comment]
I think we can wait for docker/distribution#1408 to not open second pull request.

**Comment by calavera** [comment]
agree @LK4D4

**Comment by tiborvass** [comment]
Closing, so that we dont accidentally merge before docker/distribution#1408. And I know what i'm talking about.

**Comment by thaJeztah** [comment]
https://github.com/docker/distribution/pull/1408 was merged, should we reopen @aaronlehmann ?

**Comment by aaronlehmann** [comment]
@thaJeztah Reopened with latest distribution master.

**Comment by thaJeztah** [comment]
ping @calavera @tiborvass @LK4D4 PTAL if this is ready to merge

**Comment by thaJeztah** [comment]
looks like all Janky nodes are down :cry:

**Comment by LK4D4** [comment]
Great
LGTM

**Pull request merged**